### PR TITLE
fix(autoReplace): Correct digest resolution when the replacementName and replacementVersion options are defined

### LINKED ISSUE #20304: docker: Replacement causes existing dependencies to use wrong digest
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.125.1

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

N/A

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

We're replacing an old image with a new one, but the digest for the new image is different.

This is fine and the replacement PRs look fine too, but Renovate seems to assume the replacements are also aliases.

This means that Renovate is now opening PRs for the old image with the digest for the new image...

To help illustrate this, here are our two images:

- legacy, digest: 44dea30fe85df1f3d554cee602fa69d553da7ab746c8c9f20defb1f1fc3e03b8
- current, digest: e742813303ad00bbe1409ea01eb909cb67818a8fbaae8c07df79036a7dea3e1d

Renovate is opening PRs which look like:

```diff
-FROM legacy@sha256:44dea30fe85df1f3d554cee602fa69d553da7ab746c8c9f20defb1f1fc3e03b8
+FROM legacy@sha256:e742813303ad00bbe1409ea01eb909cb67818a8fbaae8c07df79036a7dea3e1d
```

This is wrong and unexpected. I've read these issues, but they're not completely the same:

- https://github.com/renovatebot/renovate/issues/20128
- https://github.com/renovatebot/renovate/issues/19999

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

<https://github.com/renovate-reproductions/27728-docker-replacement-digest>

### PR DESCRIPTION
**Pull request by ssegato81**
## Changes

* The getDigestConfig variable uses the correct lookupName when it's running a replacement and the replacementName is set

## Context

Discussion https://github.com/renovatebot/renovate/discussions/27728
Closes https://github.com/renovatebot/renovate/issues/20304

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/process/lookup/index.ts [hunk 1]
```diff
@@ -516,6 +516,15 @@ export async function lookupUpdates(
             registryUrl: update.registryUrl ?? res.registryUrl,
             lookupName: res.lookupName,
           };
+
+          //TODO #27728
+          // If the present contains the replacementName option and this update executes the replacement
+          // uses the correct lookupname based on the new replacementName, otherwise it keeps the origina one
+          if (update.updateType === 'replacement' && config.replacementName) {
+            getDigestConfig.lookupName = config.replacementName.substring(config.replacementName.indexOf("/") + 1);
```
Comment (viceice) [inline lib/workers/repository/process/lookup/index.ts:524]: this looks wrong to me. why the `substring` ?

**Review by viceice — CHANGES_REQUESTED** [review]
needs tests

**Comment by ssegato81** [comment]
> needs tests



> needs tests

It already has a test case for that part of the code. You can see it here https://github.com/renovatebot/renovate/blob/376a2a517b1f1bcc8f0ba4606c91b8d014d0ac02/lib/workers/repository/process/lookup/index.spec.ts#L4143

I'm only improving one part of the code that fails when the replacementName option points to another image different from the original one. The test case didn't detect that error/unexpected result because it happens when the lookupUpdates function calls the getDigestConfig one with the wrong values and you are mocking that answer. I don't know if you are expecting something different.

**Comment by viceice** [comment]
you can do expectations on the function call, so it's called with the right parameters.

**Pull request closed without merging**